### PR TITLE
Dependency bumps to fix cleaninstall CI

### PR DESCRIPTION
**Pull request by mojtaba-komeili**
**Patch description**
Trying to fix some of the recently broken ParlAI tests.

### CODE REVIEW

**Comment by mojtaba-komeili** [comment]
Crowdsourcing test failure might be coming from Mephisto. 
@JackUrb it seems like Mephisto DB is trying to decode an empty JSON. Was there any recent updates in Mephisto that may cause this?
```
tests/crowdsourcing/tasks/model_chat/test_model_chat.py:159: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
parlai/crowdsourcing/utils/tests.py:264: in _test_agent_states
    agent_ids = self._register_mock_agents(num_agents=num_agents)
parlai/crowdsourcing/utils/tests.py:170: in _register_mock_agents
    agents = self.db.find_agents()
../Mephisto/mephisto/abstractions/databases/local_database.py:1076: in find_agents
    return [Agent(self, str(r["agent_id"]), row=r) for r in rows]
../Mephisto/mephisto/abstractions/databases/local_database.py:1076: in <listcomp>
    return [Agent(self, str(r["agent_id"]), row=r) for r in rows]
../Mephisto/mephisto/abstractions/providers/mock/mock_agent.py:31: in __init__
    super().__init__(db, db_id, row=row)
../Mephisto/mephisto/data_model/agent.py:73: in __init__
    self.state = AgentState(self)  # type: ignore
../Mephisto/mephisto/abstractions/blueprints/parlai_chat/parlai_chat_agent_state.py:37: in __init__
    self.load_data()
../Mephisto/mephisto/abstractions/blueprints/parlai_chat/parlai_chat_agent_state.py:72: in load_data
    state = json.load(state_json)
/usr/local/lib/python3.7/json/__init__.py:296: in load
    parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
/usr/local/lib/python3.7/json/__init__.py:348: in loads
    return _default_decoder.decode(s)
/usr/local/lib/python3.7/json/decoder.py:337: in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <json.decoder.JSONDecoder object at 0x7f326a4d8b10>, s = '', idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

/usr/local/lib/python3.7/json/decoder.py:355: JSONDecodeError
```

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 boto3==1.9.246
 botocore==1.12.246
 coloredlogs==14.0
-docutils==0.15
+docutils<0.16,>=0.14
```
Comment (stephenroller) [inline requirements.txt:4]: what version do we actually hit?

File: setup.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     reqs = []
     for line in f:
         line = line.strip()
-        reqs.append(line.split('==')[0])
```
Comment (stephenroller) [inline setup.py:25]: no, please don't do this

**Review by stephenroller — CHANGES_REQUESTED** [review]

**Comment by JackUrb** [comment]
> Was there any recent updates in Mephisto that may cause this?

This is curious as I've seen this pop up as a phantom error in the last few days, but without any changes that I've made. Possible something is going on, investigating.

File: requirements.txt (hunk 1 above)
Comment (mojtaba-komeili) [reply requirements.txt:4]: It gets the pre-release versions, something like 0.17.1b1.dev0.

File: requirements.txt [hunk 3]
```diff
@@ -35,7 +35,7 @@ scipy==1.4.1
 sh==1.12.14
 sphinx_rtd_theme==0.4.3
 sphinx-autodoc-typehints==1.10.3
-Sphinx==2.2.0
+Sphinx<4,>=2
```
Comment (mojtaba-komeili) [inline requirements.txt:38]: It was getting 4.0.0b1 for this one. Pre-release again.

**Review by stephenroller — APPROVED** [review]

**Pull request merged**